### PR TITLE
Exposed DeletionIterator and CompactMetas functions.

### PR DESCRIPTION
**Pull request by bwplotka**
Required for CLI for deletions in Thanos: https://github.com/thanos-io/thanos/pull/3421

Signed-off-by: Bartlomiej Plotka <bwplotka@gmail.com>

### CODE REVIEW

File: tsdb/compact.go [hunk 1]
```diff
@@ -330,7 +330,10 @@ func splitByRange(ds []dirMeta, tr int64) [][]dirMeta {
 	return splitDirs
 }
 
-func compactBlockMetas(uid ulid.ULID, blocks ...*BlockMeta) *BlockMeta {
+// CompactBlockMetas merges many block metas into one, combining it's source blocks together
+// and adjusting compaction level.
+// NOTE(bwplotka): Used by Thanos code.
```
Comment (roidelapluie) [inline tsdb/compact.go:335]: I don't think we need this note

File: tsdb/querier.go [hunk 2]
```diff
@@ -767,35 +768,36 @@ func (m mergedStringIter) Err() error {
...
 }
 
-// deletedIterator wraps an Iterator and makes sure any deleted metrics are not
-// returned.
-type deletedIterator struct {
-	it chunkenc.Iterator
-
-	intervals tombstones.Intervals
+// DeletedIterator wraps chunk Iterator and makes sure any deleted metrics are not returned.
+// NOTE(bwplotka): Used by Thanos code.
```
Comment (roidelapluie) [inline tsdb/querier.go:772]: idem

File: tsdb/compact.go (hunk 1 above)
Comment (bwplotka) [reply tsdb/compact.go:335]: Yea, wonder if it's not useful to not get this removed in next commit, happy to remove (:

Comment (roidelapluie) [reply tsdb/compact.go:335]: We expose many API's for many third parties;  I don't think we should mention it. I don't think we ever did.

Comment (brian-brazil) [reply tsdb/compact.go:335]: I agree. If we need this note, then that indicates this is dead code we should be deleting.

**Comment by bwplotka** [comment]
Ok (: Addressed comments. PTAL @roidelapluie @brian-brazil

**Comment by brian-brazil** [comment]
:+1:

**Review by roidelapluie — APPROVED** [review]

**Pull request merged**